### PR TITLE
Fix compatibility with Sphinx 4 and 5.0b1.

### PR DESCRIPTION
**Pull request by icemac**
Hi @stevepiercy, this PR adds up on the changes you made to `repoze.sphinx.autointerface`. Locally the tests run with Sphinx 4, 4.5 and 5.0b1.

Spoiler: I am not sure if it is right what I did, I just made the things necessary to get the tests running again.

Could you please consider this PR to make some progress for this package.

### CODE REVIEW

File: repoze/sphinx/autointerface.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def keyfunc(entry):
                 self.add_line(u'', '<autointerface>')
                 self.indent += self.content_indent
                 sourcename = u'docstring of %s.%s' % (self.fullname, name)
-                docstrings = [prepare_docstring(doc, None)]
```
Comment (icemac) [inline repoze/sphinx/autointerface.py:91]: This is needed for version 5.0b1

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
@icemac thank you! I also have no idea if it is the right thing, but as long as tests pass, I'm happy.

**Comment by stevepiercy** [comment]
Test environments are out of date and cause failures. We should drop those that are now EOL, including 3.6.

**Comment by icemac** [comment]
Hi @stevepiercy I updated the GHA config, so it should be able to run through.

**Comment by stevepiercy** [comment]
All green! Thank you!

**Pull request merged**